File: stackez.py
class Stack:
    def __init__(self):
        self.items = []
    def is_empty(self):
        return self.items == []
    def push(self,item):
        self.items.append(item)
    def pop(self):
        return self.items.pop()
    def peek(self):
        return self.items[len(self.items) - 1]
    def size(self):
        return len(self.items)
q = int(input())
st = Stack()
for i in range(q):
    s = list(map(int,input().split()))
    if s[0] == 1:
        st.push(s[1])
    elif s[0] == 2:
        if not st.is_empty():
            st.pop()
    else:
        if st.is_empty():
            print('Empty!')
        else:
            print(st.peek())

File: STACKEZ.c
#include <stdio.h>
#include <stdlib.h>

struct Node {
    long long int data;
    struct Node * next;
};
struct Node * head = NULL;
void pop(){
    if(head == NULL){ 
        return;
    }
    struct Node * temp = head;
    head = head->next;
    free(temp);
}
void push(long long int x){
    struct Node * ptr = malloc(sizeof(struct Node));
    ptr->data = x;
    ptr->next = head;
    head = ptr;
}
int isEmpty(){
    if(head == NULL) return 1;
    return 0;
}
long long int peek(){
    if(head == NULL) return -1;
    return head->data;
}

int main()
{
    int t;
    scanf("%d",&t);
    int op;
    while(t--){
        scanf("%d",&op);
        if(op == 1){
            long long int x;
            scanf("%lld",&x);
            push(x);
        }
        else if(op == 2){
            pop();
        }
        else{
            if(isEmpty()){
                printf("Empty!\n");
            }
            else{
                printf("%lld\n",peek());
            }
        }
    }
}
